crypto/rand/rand_lib.c

```diff
@@ -25,6 +25,45 @@ static CRYPTO_RWLOCK *rand_meth_lock;
 static const RAND_METHOD *default_RAND_meth;
 static CRYPTO_ONCE rand_init = CRYPTO_ONCE_STATIC_INIT;
 
+#ifdef OPENSSL_RAND_SEED_RDTSC
+/*
+ * Since we get some randomness from the low-order bits of the
+ * high-speec clock, it can help.  But don't return a status since
+ * it's not sufficient to indicate whether or not the seeding was
+ * done.
+ */
+void rand_rdtsc(void)
+{
+    unsigned char c;
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
Here is an output for the lowest 8 bit of the TSC using NIST's tool:
reading 1000000 bytes of data
Read in file rdtsc.bin, 1000000 bytes long.
Dataset: 1000000 8-bit symbols, 256 symbols in alphabet.
Output symbol values: min = 0, max = 255

Running entropic statistic estimates:
- Most Common Value Estimate: p(max) = 0.0107109, min-entropy = 6.54477
- Collision Estimate: p(max) = 0.113853, min-entropy = 3.13475
- Markov Estimate (map 6 bits): p(max) = 6.63294e-201, min-entropy = 5.19514
- Compression Estimate: p(max) = 0.17321, min-entropy = 2.52941
- t-Tuple Estimate: p(max) = 0.0172047, min-entropy = 5.86106
- LRS Estimate: p(max) = 0.146002, min-entropy = 2.77594

Running predictor estimates:
Computing MultiMCW Prediction Estimate: 100 percent complete
        Pglobal: 0.015311
        Plocal: 0.025391
MultiMCW Prediction Estimate: p(max) = 0.0253906, min-entropy = 5.29956

Computing Lag Prediction Estimate: 100 percent complete
        Pglobal: 0.043244
        Plocal: 0.300781
Lag Prediction Estimate: p(max) = 0.300781, min-entropy = 1.73321